tests/compiletest.rs

```diff
@@ -41,22 +44,32 @@ fn run_pass(path: &str) {
     compiletest::run_tests(&config);
 }
 
-fn miri_pass(path: &str, target: &str, host: &str, fullmir: bool) {
-    eprintln!("## Running run-pass tests in {} against miri for target {}", path, target);
+fn miri_pass(path: &str, target: &str, host: &str, fullmir: bool, opt: bool) {
+    let opt_str = if opt {
+        " with optimizations"
+    } else {
+        ""
+    };
+    eprintln!("## Running run-pass tests in {} against miri for target {}{}", path, target, opt_str);
     let mut config = compiletest::default_config();
     config.mode = "mir-opt".parse().expect("Invalid mode");
     config.src_base = PathBuf::from(path);
     config.target = target.to_owned();
     config.host = host.to_owned();
     config.rustc_path = PathBuf::from("target/debug/miri");
+    let mut flags = Vec::new();
     if fullmir {
         if host != target {
             // skip fullmir on nonhost
             return;
         }
         let sysroot = Path::new(&std::env::var("HOME").unwrap()).join(".xargo").join("HOST");
-        config.target_rustcflags = Some(format!("--sysroot {}", sysroot.to_str().unwrap()));
+        flags.push(format!("--sysroot {}", sysroot.to_str().unwrap()));
     }
+    if !opt {
+        flags.push("-Zmir-opt-level=0".to_owned());
+    }
```

Comment:
Good idea. Technically we should be testing all levels, but quadrupling our test times seems overkill